pkg/kv/kvserver/rangefeed/catchup_scan_test.go

```diff
@@ -149,3 +149,44 @@ func TestCatchupScanInlineError(t *testing.T) {
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "unexpected inline value")
 }
+
+func TestCatchupScanSeesOldIntent(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	// Regression test for [#85886]. When with-diff is specified, the iterator may
+	// be positioned on an intent that is outside the time bounds. When we read
+	// the intent and want to load the version, we must make sure to ignore time
+	// bounds, or we'll see a wholly unrelated version.
+	//
+	// [#85886]: https://github.com/cockroachdb/cockroach/issues/85886
+
+	ctx := context.Background()
+	eng := storage.NewDefaultInMemForTesting()
+	defer eng.Close()
+
+	// b -> version @ 1100 (visible)
+	// d -> intent @ 990   (iterator will be positioned here because of with-diff option)
+	// e -> version @ 1100
+	tsCutoff := hlc.Timestamp{WallTime: 1000} // the lower bound of the catch-up scan
+	tsIntent := tsCutoff.Add(-10, 0)          // the intent is below the lower bound
+	tsVersionInWindow := tsCutoff.Add(10, 0)  // an unrelated version is above the lower bound
+
+	require.NoError(t, storage.MVCCPut(ctx, eng, nil, roachpb.Key("b"),
+		tsVersionInWindow, hlc.ClockTimestamp{}, roachpb.MakeValueFromString("foo"), nil))
+
+	txn := roachpb.MakeTransaction("foo", roachpb.Key("d"), roachpb.NormalUserPriority, tsIntent, 100, 0)
+	require.NoError(t, storage.MVCCPut(ctx, eng, nil, roachpb.Key("d"),
+		tsIntent, hlc.ClockTimestamp{}, roachpb.MakeValueFromString("intent"), &txn))
+
+	require.NoError(t, storage.MVCCPut(ctx, eng, nil, roachpb.Key("e"),
+		tsVersionInWindow, hlc.ClockTimestamp{}, roachpb.MakeValueFromString("bar"), nil))
+
+	// Run a catchup scan across the span and watch it error.
```

Comment:
nit: we're specifically watching it _not_ error. :)

Reply:
Classic copy paste while asleep at the wheel 🤦🏽 Done